java/kafka/producer/src/main/java/KafkaProducerExample.java

```diff
@@ -74,7 +74,7 @@ public static void main(String[] args) throws InterruptedException {
                     // Increment number of sent messages only if ack is received by producer
                     numSent.incrementAndGet();
                 } catch (ExecutionException e) {
-                    log.warn("Message {} wasn't sent properly!", i);
+                    log.warn("Message {} wasn't sent properly! \n{}", i, e.getCause());
```

Comment:
Probably no need for the `\n{}`. The methods of `log` print the exception and stacktrace for an "extra" exception argument (i.e. for which there isn't a `{}` placeholder). If it's a `KafkaException` there won't be a stacktrace, but if it's not there will be, so printing it with the stacktrace is worthwhile.

Reply:
Okay, thanks for the explanation :slightly_smiling_face: